src/cascadia/TerminalSettingsEditor/Utils.h

```diff
@@ -11,63 +11,69 @@
 // setting has.
-#define INITIALIZE_BINDABLE_ENUM_SETTING(name, enumMappingsName, enumType, resourceSectionAndType, resourceProperty)                                    \
-    std::vector<winrt::Microsoft::Terminal::Settings::Editor::EnumEntry> name##List;                                                                    \
-    _##name##Map = winrt::single_threaded_map<enumType, winrt::Microsoft::Terminal::Settings::Editor::EnumEntry>();                                     \
-    auto enumMapping##name = winrt::Microsoft::Terminal::Settings::Model::EnumMappings::##enumMappingsName();                                           \
-    for (auto [key, value] : enumMapping##name)                                                                                                         \
-    {                                                                                                                                                   \
-        auto enumName = LocalizedNameForEnumName(resourceSectionAndType, key, resourceProperty);                                                        \
-        auto entry = winrt::make<winrt::Microsoft::Terminal::Settings::Editor::implementation::EnumEntry>(enumName, winrt::box_value<enumType>(value)); \
-        name##List.emplace_back(entry);                                                                                                                 \
-        _##name##Map.Insert(value, entry);                                                                                                              \
-    }                                                                                                                                                   \
-    std::sort(begin(name##List), end(name##List), EnumEntryComparator<enumType>());                                                                     \
-    _##name##List = winrt::single_threaded_observable_vector<winrt::Microsoft::Terminal::Settings::Editor::EnumEntry>(std::move(name##List));
+#define INITIALIZE_BINDABLE_ENUM_SETTING(name, enumMappingsName, enumType, resourceSectionAndType, resourceProperty)                                        \
+    do                                                                                                                                                      \
+    {                                                                                                                                                       \
+        std::vector<winrt::Microsoft::Terminal::Settings::Editor::EnumEntry> name##List;                                                                    \
+        _##name##Map = winrt::single_threaded_map<enumType, winrt::Microsoft::Terminal::Settings::Editor::EnumEntry>();                                     \
+        auto enumMapping##name = winrt::Microsoft::Terminal::Settings::Model::EnumMappings::enumMappingsName();                                             \

```

Comment:
I can only suggest turning on white space suppressing in the GitHub diff viewer (append `?w=1` to the current url).

In standard conform macros `##` can only be used to concatenate valid symbols together. You can't concatenate with `::`. This change fixes the macro, but also ensures that local variables are eagerly released to the allocator by using a do-while scope.

Reply:
The do-while loop has the advantage that it cannot be used in any other statement, for instance within the conditional statement of a while loop or if condition. It forces the programmer to treat the macro as a regular statement that has no return value and that prevents additional misuse.